rules/policy.go

```diff
@@ -83,28 +276,48 @@ func (r *DefaultRuleRenderer) ProtoRuleToIptablesRules(pRule *proto.Rule, ipVers
 				logCxt.Debug("Rule skipped.")
 				return nil
 			}
-
-			markBit, actions := r.CalculateActions(match, &ruleCopy, ipVersion)
+			if usingCIDRBlocks {
+				// The CIDR matches in the rule overflowed and we rendered them
+				// as additional rules, which set the markAllBlocksPass bit on
+				// success.  Add a match on that bit to the calculated rule.
+				match = match.MarkSet(markAllBlocksPass)
+			}
```

Comment:
Might be missing something but I _think_ it's correct; the expansion of ports works a bit differently than the new feature.  We only ever expect one of the port match rules to fire, all the rules in the port match block are checking the same thing using the same bit, i.e. whether the preceding CIDR blocks matched.

The logic is something like this:

Simplifying slightly by assuming we're only allowed 1 port in a rule and we need to match `source port  in {1234, 12345} and dest port in {80, 8080}`.  We do the cross-product of the ports:

- `(1234, 80)`
- `(1234, 8080)`
- `(12345, 80)`
- `(12345, 8080)`

and render something like this:

```
<cidr blocks>
# at this point all-blocks bit is set iff the CIDR matches were successful
if all-blocks-bit && src-port == 1234 && dest port == 80 then <rule action>
if all-blocks-bit && src-port == 1234 && dest port == 8080 then <rule action>
if all-blocks-bit && src-port == 12345 && dest port == 80 then <rule action>
if all-blocks-bit && src-port == 12345 && dest port == 8080 then <rule action>
```

(In reality, the kernel can support 15 ports in a rule so we do the cross-product of chunks of 15 ports but the logic is similar.)